### DISCUSSION TITLE
Customize API endpoint could not work

### DISCUSSION THREAD

**Discussion by TianzeWang** [danny-avila/LibreChat#4156] [Troubleshooting]
### What happened?

I set up my customize API, which is based on a cloudflare service routing the Google vertex Claude API, and I have my baseurl and the api_key I defined. I set up the same in yaml and .env, and the endpoint could show in dropdown list, but it will pop error. 
<img width="703" alt="image" src="https://github.com/user-attachments/assets/e680525f-0e64-4c7e-9644-6279327094a1">

I checked the log:

{
    "error": {
        "message": "invalid x-api-key",
        "type": "authentication_error"
    },
    "headers": {
        "access-control-allow-origin": "*",
        "cf-ray": "8c5fbc08efd2d1be-KIX",
        "connection": "keep-alive",
        "content-length": "86",
        "content-type": "application/json",
        "date": "Fri, 20 Sep 2024 06:24:47 GMT",
        "nel": "{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}",
        "report-to": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v4?s=Focmn0%2FKf3nsCPf%2BPx5%2Ff6xNLWgZ23rSnpqHC4yzId7KHvOE7L7r%2F5pN86xtH4bI8ufAY3DPB49n1RmR%2FKl5Yqvpbe8W8RjnKdMcGMkSl2fIf49zvD%2BA0PUQiMZAFADKCILaAEdnjodEXz0NOp6YCz4%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}",
        "server": "cloudflare",
        "vary": "Accept-Encoding"
    },
    "level": "error",
    "message": "[handleAbortError] AI response error; aborting request: 401 invalid x-api-key",
    "stack": "Error: 401 invalid x-api-key\n    at APIError.generate (/app/api/node_modules/openai/error.js:48:20)\n    at OpenAI.makeStatusError (/app/api/node_modules/openai/core.js:263:33)\n    at OpenAI.makeRequest (/app/api/node_modules/openai/core.js:306:30)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async ChatCompletionStream._createChatCompletion (/app/api/node_modules/openai/lib/ChatCompletionStream.js:53:24)\n    at async ChatCompletionStream._runChatCompletion (/app/api/node_modules/openai/lib/AbstractChatCompletionRunner.js:314:16)",
    "status": 401,
    "type": "authentication_error"
}

### Steps to Reproduce

1. You might need to use my personal API and endpoint. Please reach out me and I will share.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by TianzeWang** [comment]
Updating the curl command that could work correctly

curl -X POST "https://cld.wangtianze3682.workers.dev/v1/messages" \
  -H "x-api-key: MYAPIKEY" \
  -H "anthropic-version: 2023-06-01" \
  -H "Content-Type: application/json" \
  -d '{
    "model": "claude-3-5-sonnet",
    "max_tokens": 4096,
    "messages": [
      {"role": "user", "content": "Your test message here"}
    ]
  }'

**Comment by danny-avila** [comment]
Not a bug, this is a configuration issue. See your error, you're getting an authentication issue.

**Comment by TianzeWang** [comment]
I think I have followed the instructions. Really appreciate if Danny could help point out where I am wrong @danny-avila 
1. in the **librechat.yaml** file:
<img width="636" alt="image" src="https://github.com/user-attachments/assets/fd35f4ab-7312-4a2d-9e96-6680b50a70bb">

2. In the **.env** file I configured VERTEX_CLAUDE_API_KEY=my own key, without quotations
<img width="337" alt="image" src="https://github.com/user-attachments/assets/a42071ec-f7fc-42c4-97e6-50a394b30810">

3. In the **docker-compose.override.yml**, I set
<img width="466" alt="image" src="https://github.com/user-attachments/assets/07515fa6-d056-41cd-a1fe-51ba06d3370a">
